test/parallel/test-assert.js

```diff
@@ -566,84 +638,76 @@ test('Test strict assert', () => {
   Error.stackTraceLimit = tmpLimit;
 
   // Test error diffs.
-  let message = [
-    start,
-    `${actExp} ... Lines skipped`,
-    '',
-    '  [',
-    '    [',
-    '      [',
-    '        1,',
-    '        2,',
-    '+       3',
-    "-       '3'",
-    '      ]',
-    '...',
-    '    4,',
-    '    5',
-    '  ]'].join('\n');
+  let message = 'Expected values to be strictly deep-equal:\n' +
+    '+ actual - expected\n' +
+    '\n' +
+    '  [\n' +
+    '    [\n' +
+    '      [\n' +
+    '        1,\n' +
+    '        2,\n' +
+    '+       3\n' +
+    "-       '3'\n" +
+    '      ]\n' +
+    '    ],\n' +
+    '    4,\n' +
+    '    5\n' +
+    '  ]\n';
   strict.throws(
     () => strict.deepEqual([[[1, 2, 3]], 4, 5], [[[1, 2, '3']], 4, 5]),
     { message });
 
-  message = [
-    start,
-    `${actExp} ... Lines skipped`,
-    '',
-    '  [',
-    '    1,',
-    '...',
-    '    1,',
-    '    0,',
-    '-   1,',
-    '    1,',
-    '...',
-    '    1,',
-    '    1',
-    '  ]',
-  ].join('\n');
+  message = 'Expected values to be strictly deep-equal:\n' +
+  '+ actual - expected\n' +
+  '... Skipped lines\n' +
+  '\n' +
+  '  [\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '...\n' +
+  '-   1\n' +
+  '  ]\n';
   strict.throws(
     () => strict.deepEqual(
       [1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1],
       [1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1]),
     { message });
 
-  message = [
-    start,
-    `${actExp} ... Lines skipped`,
-    '',
-    '  [',
-    '    1,',
-    '...',
-    '    1,',
-    '    0,',
-    '+   1,',
-    '    1,',
-    '    1,',
-    '    1',
-    '  ]',
-  ].join('\n');
+  message = 'Expected values to be strictly deep-equal:\n' +
+  '+ actual - expected\n' +
+  '... Skipped lines\n' +
+  '\n' +
+  '  [\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '    1,\n' +
+  '    0,\n' +
+  '    1,\n' +
+  '...\n' +
```

Comment:
I believe the former implementation always printed at least one nop line before a type change. That helps to identify the correct part. I am fine to land this even without, but I think it would be good to keep that behavior.